### PR TITLE
Fix typos found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
# Description

Fix typos found by [codespell](https://github.com/codespell-project/codespell).

# Testing

\-

# Checklist

 - [ ] added tests
 - [x] updated/added doxygen
 - [x] updated appropriate READMEs
 - [x] Updated manual and documentation

### CODE REVIEW

File: IDE/IAR-EWARM/Projects/benchmark/wolfCrypt-benchmark.ewp [hunk 1]
```diff
@@ -937,7 +937,7 @@
         </option>
         <option>
           <name>IarchiveOutput</name>
-          <state>###Unitialized###</state>
+          <state>###Uninitialized###</state>
```
Comment (DimitriPapadopoulos) [inline IDE/IAR-EWARM/Projects/benchmark/wolfCrypt-benchmark.ewp:940]: Not sure about this series of fixes: Is `Unitialized` a documented misspelling of EWP files?

File: wolfcrypt/src/kdf.c [hunk 2]
```diff
@@ -129,7 +129,7 @@ int wc_PRF(byte* result, word32 resLen, const byte* secret,
     if (lastLen)
         times += 1;
 
-    /* times == 0 iif resLen == 0, but times == 0 abides clang static analyzer
+    /* times == 0 if resLen == 0, but times == 0 abides clang static analyzer
```
Comment (DimitriPapadopoulos) [inline wolfcrypt/src/kdf.c:132]: Did you mean “iff” (if and only if)?

**Comment by JacobBarthelmeh** [comment]
Thanks @DimitriPapadopoulos!

I do not see you on our contributor list. Could you reach out to the support @ wolfssl email address for getting a contributor agreement?

Regards,
Jacob

**Comment by embhorn** [comment]
@DimitriPapadopoulos is approved as a wolfSSL contributor

File: wolfcrypt/src/ge_448.c
Comment (embhorn) [inline wolfcrypt/src/ge_448.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/kdf.c (hunk 2 above)
Comment (embhorn) [reply wolfcrypt/src/kdf.c:132]: "if" is correct here

File: wolfcrypt/src/poly1305_asm.S
Comment (embhorn) [inline wolfcrypt/src/poly1305_asm.S:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/port/arm/armv8-sha512-asm.S
Comment (embhorn) [inline wolfcrypt/src/port/arm/armv8-sha512-asm.S:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/port/arm/armv8-sha512-asm_c.c
Comment (embhorn) [inline wolfcrypt/src/port/arm/armv8-sha512-asm_c.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: IDE/IAR-EWARM/Projects/benchmark/wolfCrypt-benchmark.ewp (hunk 1 above)
Comment (DimitriPapadopoulos) [reply IDE/IAR-EWARM/Projects/benchmark/wolfCrypt-benchmark.ewp]: Indeed, I don't know what these files are, but *.ewp files:
```
$ grep -R '>###Uninitialized###'
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/Projects/lib/wolfSSL-Lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/Projects/lib/wolfSSL-Lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/Projects/lib/wolfSSL-Lib.ewd:          <state>###Uninitialized###</state>
IDE/IAR-EWARM/Projects/lib/wolfSSL-Lib.ewd:          <state>###Uninitialized###</state>
$ 
```
differ from *.ewd files:
```
$ grep -R '>###Unitialized###'
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_lib_SAMV71_XULT/wolfcrypt_lib.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_benchmark_SAMV71_XULT/wolfcrypt_benchmark.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/embOS/SAMV71_XULT/embOS_wolfcrypt_test_SAMV71_XULT/wolfcrypt_test.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/Projects/test/wolfCrypt-test.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/Projects/benchmark/wolfCrypt-benchmark.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/Projects/lib/wolfSSL-Lib.ewp:          <state>###Unitialized###</state>
IDE/IAR-EWARM/Projects/lib/wolfSSL-Lib.ewp:          <state>###Unitialized###</state>
$ 
```

File: wolfcrypt/src/sp_arm32.c
Comment (embhorn) [inline wolfcrypt/src/sp_arm32.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_arm64.c
Comment (embhorn) [inline wolfcrypt/src/sp_arm64.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_armthumb.c
Comment (embhorn) [inline wolfcrypt/src/sp_armthumb.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_c32.c
Comment (embhorn) [inline wolfcrypt/src/sp_c32.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_cortexm.c
Comment (embhorn) [inline wolfcrypt/src/sp_cortexm.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_dsp32.c
Comment (embhorn) [inline wolfcrypt/src/sp_dsp32.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_int.c
Comment (embhorn) [inline wolfcrypt/src/sp_int.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_x86_64.c
Comment (embhorn) [inline wolfcrypt/src/sp_x86_64.c:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_x86_64_asm.S
Comment (embhorn) [inline wolfcrypt/src/sp_x86_64_asm.S:1]: Please remove this from review. It will be fixed in a separate PR.

File: wolfcrypt/src/sp_x86_64_asm.asm
Comment (embhorn) [inline wolfcrypt/src/sp_x86_64_asm.asm:1]: Please remove this from review. It will be fixed in a separate PR.

**Comment by embhorn** [comment]
Hi @DimitriPapadopoulos 

Looking good! The generated file spelling fixes have been merged. 

Some of the files have been removed, so please update the PR with those changes.

**Comment by DimitriPapadopoulos** [comment]
I have rebased against _master_.

File: tests/api.c [hunk 3]
```diff
@@ -19134,6 +19134,7 @@ static int test_wc_curve25519_export_key_raw(void)
     ExpectIntEQ(wc_curve25519_export_key_raw(&key, privateKey, &prvkSz,
         publicKey, NULL), BAD_FUNC_ARG);
 
+<<<<<<< HEAD
```
Comment (embhorn) [inline tests/api.c:19137]: There are merge artifacts that need to be resolved. This is why the tests are failing.

**Review by embhorn — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 3 above)
Comment (DimitriPapadopoulos) [reply tests/api.c]: Ah, sorry about that. Should be fixed.

File: wolfcrypt/src/fe_448.c
Comment (embhorn) [inline wolfcrypt/src/fe_448.c:1]: These should be resolved by #6543

File: wolfcrypt/src/poly1305_asm.S [hunk 4]
```diff
@@ -324,7 +324,7 @@ L_poly1305_avx_final_no_more:
         movq	%rcx, %r8
         andq	$3, %rcx
         shrq	$2, %r8
-        #   Multiply by 5
+        #   Multily by 5
```
Comment (embhorn) [inline wolfcrypt/src/poly1305_asm.S:327]: This is not a valid fix.

File: wolfcrypt/src/poly1305_asm.S
Comment (embhorn) [inline wolfcrypt/src/poly1305_asm.S:1]: Can you just remove this file from the review?

File: wolfcrypt/src/port/arm/armv8-sha512-asm_c.c
Comment (embhorn) [inline wolfcrypt/src/port/arm/armv8-sha512-asm_c.c:1]: Please remove this file from the review.

File: wolfcrypt/src/sp_arm32.c
Comment (embhorn) [inline wolfcrypt/src/sp_arm32.c:1]: Please remove this file from the review.

File: wolfcrypt/src/sp_arm64.c
Comment (embhorn) [inline wolfcrypt/src/sp_arm64.c:1]: Please remove this file from the review.

File: wolfcrypt/src/sp_armthumb.c
Comment (embhorn) [inline wolfcrypt/src/sp_armthumb.c:1]: Please remove this file from the review.

File: wolfcrypt/src/sp_c32.c
Comment (embhorn) [inline wolfcrypt/src/sp_c32.c:1]: Please remove this file from the review.

File: wolfcrypt/src/sp_x86_64_asm.S
Comment (embhorn) [inline wolfcrypt/src/sp_x86_64_asm.S:1]: Please remove this file from the review.

File: wolfcrypt/src/sp_x86_64_asm.asm
Comment (embhorn) [inline wolfcrypt/src/sp_x86_64_asm.asm:1]: Please remove this file from the review.

**Review by embhorn — CHANGES_REQUESTED** [review]
The files touched by #6543 need to be removed from this PR, not reverted.

File: wolfcrypt/src/poly1305_asm.S
Comment (DimitriPapadopoulos) [reply wolfcrypt/src/poly1305_asm.S:1]: Reverting the change is the same as removing the file. I'll just squash the commits.

**Comment by DimitriPapadopoulos** [comment]
Remove [wolfcrypt/src/port/arm/armv8-sha512-asm.S](https://github.com/wolfSSL/wolfssl/pull/6506/files#diff-ee83aa46cc4099e06e2699c521adf764f2c9858c3e8cf22a9cdea7bf6a41d936) too?

**Comment by embhorn** [comment]
> Shouldn't I remove `wolfcrypt/src/port/arm/armv8-sha512-asm.S` too?

Yes, I missed that in my comments.

File: wolfcrypt/src/port/arm/armv8-sha512-asm.S
Comment (embhorn) [inline wolfcrypt/src/port/arm/armv8-sha512-asm.S:1]: Please remove this file from the review

**Review by embhorn — CHANGES_REQUESTED** [review]
After removing that last assembly file I think it will be ready for final review and merge. Thanks for your patience and perseverance!

File: IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md [hunk 5]
```diff
@@ -334,4 +334,4 @@ wolfSSL_CTX_use_certificate_buffer あるいはwolfSSL_CTX_use_certificate_chain
 上記制限1 ~ 4は次版以降のTSIPによって改善が見込まれています。
 
 ## 11. サポート
-ご質問・ご要望は、info@wolfssl.jp まで日本語でお知らせください。
\ No newline at end of file
+ご質問・ご要望は、info@wolfssl.jp まで日本語でお知らせください。
```
Comment (embhorn) [inline IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md:337]: What changed here?

Comment (DimitriPapadopoulos) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md:337]: Good question. I assumed it is a DOS line ending in a POSIX file, or something similar, but let me check.

Comment (DimitriPapadopoulos) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md:337]: Strangely [Gnome Meld](https://meldmerge.org/) doesn't show any difference on that line. However, given this is the last line, I guess it's a missing line break at the end of that last line. My editor is POSIX compliant and always adds a line break at end-of-line, including the last line. Let me double-check, though.

Comment (DimitriPapadopoulos) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md:337]: Indeed:
```
$ env LANG=C diff README_JP.orig README_JP.md
240c240
< 実行すると、下記のようなメッセージがRenesas Debug Virtual Consol に表示されます。
---
> 実行すると、下記のようなメッセージがRenesas Debug Virtual Console に表示されます。
301c301
< Renesas Debug Virtual Consol に下記のように表示されていれば、ボード上の TLS サーバーとクライアント間でTLS通信が行われています。
---
> Renesas Debug Virtual Console に下記のように表示されていれば、ボード上の TLS サーバーとクライアント間でTLS通信が行われています。
337c337
< ご質問・ご要望は、info@wolfssl.jp まで日本語でお知らせください。
\ No newline at end of file
---
> ご質問・ご要望は、info@wolfssl.jp まで日本語でお知らせください。
$ 
````
Would you like me to to remove that last line break, or do you prefer your files to be POSIX-compliant?

Comment (embhorn) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md:337]: Please leave it unchanged.

Comment (DimitriPapadopoulos) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md]: All the editors I tried ensure the presence of a line break at the end of every line, including the last line. I ended up removing the last character with:
```
sed '$ s/.$//' IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md
```

Comment (DimitriPapadopoulos) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md]: Nope, sed removed the  `。` in addition to the line break. I need to find something else.

Comment (DimitriPapadopoulos) [reply IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md]: This did the trick:
```
truncate -s -1 IDE/Renesas/e2studio/RX72N/EnvisionKit/Simple/README_JP.md
```

**Review by embhorn — DISMISSED** [review]
I think we are all set with this PR!

**Comment by embhorn** [comment]
Jenkins retest this please

**Comment by embhorn** [comment]
Hi @DimitriPapadopoulos 

We had some testing issues this week. Could you please rebase to fix the `asn.c` issue and then we'll get this PR into final review.

Thanks,
Eric

**Comment by DimitriPapadopoulos** [comment]
Done.

Perhaps because of dictionary updates, I have found these additional typos:
* `wolfcrypt/src/sp_dsp32.c` mulitplier → multiplier / co-ordinate → coordinate / co-ordinates → coordinates
* `wolfcrypt/src/sp_int.c` multipling → multiplying / mulitply → multiply
* `wolfcrypt/src/sp_arm64.c` higest → highest
* `wolfcrypt/src/ge_448.c` indeces → indices

These files do not look like they are generated. Is it OK to fix them?

**Comment by embhorn** [comment]
> Done.
> 
> I have found these residual typos:
> 
> * `wolfcrypt/src/sp_dsp32.c` mulitplier → multiplier | co-ordinate → coordinate | co-ordinates → coordinates
> * `wolfcrypt/src/sp_int.c` multipling → multiplying | mulitply → multiply | Iniitialize → Initialize | respresentation → representation | valus→ values | likey → likely | Euclidian → Euclidean
> * `wolfcrypt/src/sp_arm64.c` higest → highest
> * `wolfcrypt/src/ge_448.c` indeces → indices
> 
> I seem to recall these files are generated by scripts, it looks like not all typos have been fixed.

Hi Dimitri

Please remove `wolfcrypt/src/sp_arm64.c` and `wolfcrypt/src/ge_448.c` from the PR.

**Comment by DimitriPapadopoulos** [comment]
I have rebased.

I don't see `wolfcrypt/src/sp_arm64.c` and `wolfcrypt/src/ge_448.c` in the PR. They’re part of the automatically generated files you need to fix on your side.

**Comment by embhorn** [comment]
I fixed these two in the scripts repository:
> * `wolfcrypt/src/sp_arm64.c` higest → highest
> * `wolfcrypt/src/ge_448.c` indeces → indices

The others are not showing up in the scripts, so they probably only exist in the wolfSSL source files.

**Comment by embhorn** [comment]
@SparkiDev - Are `wolfcrypt/src/sp_dsp32.c` and `wolfcrypt/src/sp_int.c` generated files?

**Review by embhorn — APPROVED** [review]
Issues in sp_int.c will be addressed in another PR.

**Comment by embhorn** [comment]
Passing over to @JacobBarthelmeh for final review / merge

**Review by JacobBarthelmeh — APPROVED** [review]
